datastore/dsiface/dsiface_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dsiface
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"math/rand"
+	"os"
+	"testing"
+
+	"cloud.google.com/go/datastore"
+)
+
+func TestIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("integration tests skipped in short mode")
+	}
+
+	if os.Getenv("DATASTORE_PROJECT_ID") == "" || os.Getenv("GOOGLE_APPLICATION_CREDENTIALS") == "" {
```

Comment:
Right, but the reason why storage also isn't asking specifically requiring `GOOGLE_APPLICATION_CREDENTIALS` is because that's a variable that's being checked by the client library.

Also, `GOOGLE_APPLICATION_CREDENTIALS` is not actually required. It's one of several (four) ways you can authenticate without passing [an option](https://godoc.org/google.golang.org/api/option) to the client library: [see here](https://github.com/golang/oauth2/blob/master/google/default.go#L66-L76).

Reply:
Would you suggest then that I only guard DATASTORE_PROJECT_ID? Or remove the guard all together and let the test fail if it's not configured?

Reply:
Would you suggest then that I only guard DATASTORE_PROJECT_ID? Or remove the guard all together and let the test fail if it's not configured?

Reply:
I'd just remove all references of it entirely. If you add a README.md that describes how to run these tests, that would be an appropriate place to put it, or you might link to https://godoc.org/cloud.google.com/go#hdr-Authentication_and_Authorization.

Reply:
That sounds good. I added a README giving quick examples on how to run the tests with and without configuration.